platform/posix/transport/src/openssl_posix.c

```diff
@@ -484,6 +486,10 @@ OpensslStatus_t Openssl_Connect( NetworkContext_t * pNetworkContext,
             LogError( ( "SSL_new failed to create a new SSL context." ) );
             returnStatus = OPENSSL_API_ERROR;
         }
+        else
```

Comment:
If the ssl context is NULL, then the returnStatus is error and we skip all other if cases.

Reply:
I didn't understand your point here..